### PR TITLE
Change permissions for agency settings

### LINKED ISSUE #240: Modify permissions for Agency Settings page
Previously I asked @ivan-kishko to hide the Agency Settings page for everyone unless they had RECIDIVIZ_ADMIN permissions. I was wrong about this. We want to always show the page, BUT we want to hide certain sections from non-admins. The logic should be:

* Everyone can see the Agency Settings page in the Navigation and can load this page
* Everyone can see the Basic Information section
* All Supervision agencies can see the Supervision Populations section
* Recidiviz Admins only can see the Team Management section
* Recidiviz Admins only can see the Jurisdictions section

Sorry about this change. Thank you!!

### PR DESCRIPTION
**Pull request by ivan-kishko**
## Description of the change

> Change permissions of what non admins can view about agency settings

## Type of change

Type: Non-breaking refactor

> All pull requests must have at least one of the following labels applied (otherwise the PR will fail):

| Label                       	| Description                                                                                               	|
|-----------------------------	|-----------------------------------------------------------------------------------------------------------	|
| Type: Bug                   	| non-breaking change that fixes an issue                                                                   	|
| Type: Feature               	| non-breaking change that adds functionality                                                               	|
| Type: Breaking Change       	| fix or feature that would cause existing functionality to not work as expected                            	|
| Type: Non-breaking refactor 	| change addresses some tech debt item or prepares for a later change, but does not change functionality    	|
| Type: Configuration Change  	| adjusts configuration to achieve some end related to functionality, development, performance, or security 	|
| Type: Dependency Upgrade      | upgrades a project dependency - these changes are not included in release notes                             |

## Related issues

Closes #240 

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Review by lilidworkin — APPROVED** [review]
Looks great! Let me know if you're able to test it out using the special non-admin account I shared with you over Slack. Thanks!

File: publisher/src/components/Settings/AgencySettings.tsx [hunk 1]
```diff
@@ -166,25 +168,27 @@ export const AgencySettings: React.FC = observer(() => {
...
+            <AgencySettingsBlockDescription>
+              These are the other people at your agency who have accounts on
+              Publisher. If there is someone you work with who you think should
+              be on Publisher, contact the Justice Counts team at{" "}
+              <a href="mailto:justice-counts-support@csg.org">
+                justice-counts-support@csg.org
+              </a>
+              .
+            </AgencySettingsBlockDescription>
+            {agencyTeam?.map(({ name }) => (
```
Comment (terryttsai) [inline publisher/src/components/Settings/AgencySettings.tsx:183]: Should we allow non-admins to still be able to view the list of users on their team? @lilidworkin

**Review by terryttsai — APPROVED** [review]
Looks good to me!

**Comment by ivan-kishko** [comment]
> Looks great! Let me know if you're able to test it out using the special non-admin account I shared with you over Slack. Thanks!

I just used isAdmin =  false to test, it is ok

File: publisher/src/components/Settings/AgencySettings.tsx (hunk 1 above)
Comment (terryttsai) [reply publisher/src/components/Settings/AgencySettings.tsx:183]: Question answered here: https://github.com/Recidiviz/justice-counts/issues/240#issuecomment-1360226140

**Pull request merged**